config/config.go

```diff
@@ -625,10 +626,23 @@ func (c *RemoteWriteConfig) UnmarshalYAML(unmarshal func(interface{}) error) err
 		}
 	}
 
-	// The UnmarshalYAML method of HTTPClientConfig is not being called because it's not a pointer.
-	// We cannot make it a pointer as the parser panics for inlined pointer structs.
-	// Thus we just do its validation here.
-	return c.HTTPClientConfig.Validate()
+	return c.Validate()
+}
+
+func (c *RemoteWriteConfig) Validate() error {
+	if err := c.HTTPClientConfig.Validate(); err != nil {
+		return err
+	}
+
+	httpClientConfigEnabled := c.HTTPClientConfig.BasicAuth != nil ||
+		len(c.HTTPClientConfig.BearerToken) > 0 ||
+		len(c.HTTPClientConfig.BearerTokenFile) > 0
+
+	if httpClientConfigEnabled && c.SigV4Config.Enabled {
+		return fmt.Errorf("at most one of basic_auth & bearer_token & bearer_token_file & sigv4 must be configured")
+	}
+
+	return nil
```

Comment:
I'm not thrilled about how this works, would like suggestions for better ways of doing this.